### DISCUSSION TITLE
Consistency for frontend URIs

### DISCUSSION THREAD

**Discussion by MoritzWeber0** [DSD-DBS/capella-collab-manager#233] [General]
We should bring some consistency into our route URIs. Here is my proposal (I'm happy to hear your opinion about it): 

- `/resources` for the resource overview (e.g. `/projects` for the projects overview)
-  `/resources/action` for a specific action (not related to a specific instance of the resource), e.g. `/projects/create` to create a project and `/project/:name/models/create` to create a model in the project with name `:name`.
-  `/resources/action/subaction` for a specific action related to a parent action, e.g. `/project/:name/models/create/source` to go to the source stage of the model creation. 
- `resource/:name` to access a single instance of type resource.

**Comment by amolenaar** [comment]
The naming scheme looks pretty straightforward.

I have some doubts about the 3rd point. Since a "subaction" happens as part of a bigger action (e.g. create a new project). That would mean, that all information gathered in previous steps should be carried along. Either on the URL, or stored somewhere in javascript. The first case is somewhat tricky IMHO, and the second case has some funny downsides as well (try reload the page, for example).

**Reply by MoritzWeber0** [reply]
We should try to keep the subactions independent from each other (as far as possible). When we look at the creation of projects, we don't use the subactions URL scheme (because we use a stepper for it). But let's imagine that we would use one subaction URL for each of the two steps: 

1. Provide metadata (name, description, etc.) - URL: `/projects/create/metadata`
2. Add users - URL: `/projects/create/users`

When we go to the first step, there is no information passed. So no issue when using the first step. 
The second step gets interesting now, as we need the project name. In this simple case, I'd use something like `/project/:slug/create/users` for it. The project name would be passed via the URL and the process can be continued (even in the case of a refresh) by fetching the project information from the backend.
